src/Microsoft.Restier.EntityFramework/Model/ModelMapper.cs

```diff
@@ -60,10 +61,10 @@ public ModelMapper(Type dbContextType)
             if (property != null)
             {
                 var type = property.PropertyType;
-                if (type.IsGenericType &&
-                    type.GetGenericTypeDefinition() == typeof(DbSet<>))
+                var parent = GetGenericParent(type);
```

Comment:
It can be used, however implementation of the `TypeExtensions.FindGenericType` seems a bit costly in terms of computational complexity. Say the type represented by the parameter `type` is at level `h` in its inheritance tree (distance from the type of `object`) and there are `n` interfaces implemented by it. Here are some cases:

- If the parameter `definition` does not represent an interface, then the complexity of the method would be `O(h)`,
- If the parameter `definition` represents an interface implemented by `type`, then the complexity of the method would be `O(n)`
- However, if the parameter `definition` represents an interface which is not implemented by `type`, then the complexity would be `O(n h)` which represents the worst case complexity of this method.

To make the worst case complexity `O(max(n, h))`, at least the inner `foreach` loop which checks the `definition` against the implemented interfaces should be moved out of the while loop.

If it is OK for you, i can add the changes proposed above and for its usage in `ModelMapper` to this pull request.